packages/agent-core/src/services/mcp-manager.ts

```
@@ -288,12 +349,62 @@ export class MCPManager implements IMCPManager {
 
+  async updateAuthToken(token: string | null): Promise<void> {
+    logger.info(`Updating auth token: ${token ? "present" : "null"}`);
+    this.authToken = token;
+    this.orchestrator.setAuthToken(token);
+
+    // Handle RAG connection based on token
+    if (token) {
+      // Try to connect to RAG server if we have the config
+      const ragConfig = createMCPServerConfig("rag", process.env as any);
+
+      if (ragConfig && ragConfig.url) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)